<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="./img/favicon.ico">

	<title>Botanyhelp</title>

        <link href="./css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="./css/font-awesome-4.0.3.css" rel="stylesheet">
        <link href="./css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="./css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body class="homepage">

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            

            <!-- Main title -->
            <a class="navbar-brand" href=".">Botanyhelp</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#education">Education</a></li>
        
    
        <li class="main "><a href="#experience">Experience</a></li>
        
    
        <li class="main "><a href="#certificates">Certificates</a></li>
        
    
        <li class="main "><a href="#programming">Programming</a></li>
        
            <li><a href="#c">C</a></li>
        
            <li><a href="#java">Java</a></li>
        
            <li><a href="#javascript">Javascript</a></li>
        
            <li><a href="#perl">Perl</a></li>
        
            <li><a href="#php">Php</a></li>
        
            <li><a href="#python">Python</a></li>
        
            <li><a href="#r">R</a></li>
        
            <li><a href="#ruby">Ruby</a></li>
        
            <li><a href="#web-content">Web Content</a></li>
        
            <li><a href="#cloud">Cloud</a></li>
        
            <li><a href="#databases">Databases</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="education">Education</h2>
<ul>
<li><strong>B.S. Chemical Engineering, Penn State 1996</strong>. Total transformation in 4.5 years. </li>
</ul>
<h2 id="experience">Experience</h2>
<ul>
<li>(05/2018 - 02/2021) Penn State Biology - State College, PA </li>
<li>(09/2016 - 06/2017) McCrossin - Bellefonte, PA </li>
<li>(07/2010 - 09/2016) Center For Craniofacial and Dental Genetics - Pittsburgh, PA </li>
<li>(01/2006 - 07/2010) University of Wisconsin Botany - Madison, WI </li>
<li>(08/2004 - 06/2005) Penn State Computer Science and Engineering - State College, PA </li>
<li>(04/2001 - 07/2003) University of Wisconsin Family Medicine - Madison, WI </li>
<li>(01/1998 - 04/2000) Information Management Consultants, Inc. - McLean, VA </li>
<li>(12/1996 - 07/1997) Ye Olde College Diner - State College, PA</li>
</ul>
<h2 id="certificates">Certificates</h2>
<p>Enrolled in and completed these online <a href="http://www.edx.org">edX Platform</a> courses during 2015 and 2016.  The R programming, statistics and biology courses were presented by <strong>Rafael Irizarry, Professor of Biostatistics, Harvard T.H. Chan School of Public Health</strong>.  Super-Earths and Life was presented by <strong>Dimitar Sasselov, Professor of Astronomy, Harvard University</strong>.  The Einstein Revolution was presented by <b>Peter Galison Professor in History of Science and Physics Harvard University</b>.  The biostatistics courses are based on this book, available free: <a href="https://leanpub.com/dataanalysisforthelifesciences">Data Analysis for the Life Sciences, Rafael A Irizarry and Michael I Love</a> and also here on github: <a href="https://github.com/genomicsclass">github.com/genomicsclass</a>.  I completed six of these courses on time and with scores high enough to earn a certificate.  For two classes, ChIP-seq data analysis and DNA methylation data analysis, scores were too low to earn a certificate.  Detailed performance information are shown below for all completed courses.  </p>

<ul>
<li>
<p>PH525.1x: Statistics and R for the Life Sciences <a href="./images/PH525_1xProgressedX_1.pdf">pdf</a> <a href="./images/PH525_1xProgressedX_1.png">png</a></p>
</li>
<li>
<p>PH525.2x: Introduction to Linear Models and Matrix Algebra <a href="./images/PH525_2xProgressedX_2.pdf">pdf</a> <a href="./images/PH525_2xProgressedX_2.png">png</a></p>
</li>
<li>
<p>PH525.4x: Introduction to Bioconductor <a href="./images/PH525_4xProgressedX_4.pdf">pdf</a> <a href="./images/PH525_4xProgressedX_4.png">png</a></p>
</li>
<li>
<p>PH525.5x Case Study: RNA-seq data analysis <a href="./images/PH525_5xProgressedX_5.pdf">pdf</a> <a href="./images/PH525_5xProgressedX_5.png">png</a></p>
</li>
<li>
<p>PH525.6x Case Study: Variant Discovery and Genotyping <a href="./images/PH525_6xProgressedX_6.pdf">pdf</a> <a href="./images/PH525_6xProgressedX_6.png">png</a></p>
</li>
<li>
<p>PH525.7x Case Study: ChIP-seq data analysis <a href="./images/PH525_7xProgressedX_7.pdf">pdf</a> <a href="./images/PH525_7xProgressedX_7.png">png</a></p>
</li>
<li>
<p>PH525.8x Case study: DNA methylation data analysis <a href="./images/PH525_8xProgressedX_8.pdf">pdf</a> <a href="./images/PH525_8xProgressedX_8.png">png</a></p>
</li>
<li>
<p>SPU30x: Super-Earths and Life <a href="./images/SPU30xProgressedX.pdf">pdf</a> <a href="./images/SPU30xProgressedX.png">png</a></p>
</li>
<li>
<p>EMC2x: The Einstein Revolution <a href="./images/EMC2x_Progress_edX.pdf">pdf</a> <a href="./images/EMC2x_Progress_edX.png">png</a></p>
</li>
</ul>
<h2 id="programming">Programming</h2>
<h3 id="c">C</h3>
<ul>
<li>
<p><strong>Deadtapes iPhone App</strong> is available on the Apple App Store: <a href="https://itunes.apple.com/us/app/deadtapes/id838487529?mt=8">https://itunes.apple.com/us/app/deadtapes/id838487529?mt=8</a></p>
</li>
<li>
<p><strong>Nuclides iPhone App</strong>  Nuclides was constructed using Objective-C in xCode using iOS7 in February 2014.  This application behaves very similarly to the Nuclides Android app available on Google Play.  The Nuclides iPhone App is not available on the Apple App Store.  <a href="https://github.com/botanyhelp/Nuclides_ios_objc">(code on github)</a></p>
</li>
</ul>
<h3 id="java">Java</h3>
<ul>
<li>
<p>Many android apps <a href="https://play.google.com/store/apps/developer?id=Thomas+Maher">https://play.google.com/store/apps/developer?id=Thomas+Maher</a></p>
</li>
<li>
<p><strong>Herbjava tomcat servlet application</strong>.  When running, The Wisconsin Herbarium java port will perform very similarly to the existing perl, python and off-the-shelf web servers running in three other places: 
<!--TTT20210318 <a href="http://www.botany.wisc.edu/wisflora/">http://www.botany.wisc.edu/wisflora/</a>, <a href="http://wisconsinherbarium.appspot.com/">http://wisconsinherbarium.appspot.com/</a>, <a href="http://herbarium.wisc.edu/">http://herbarium.wisc.edu/</a>,-->  <a href="https://github.com/botanyhelp/Herbjava">(code on github)</a></p>
</li>
</ul>
<h3 id="javascript">Javascript</h3>
<ul>
<li>
<p><strong>Awesomplete javascript autocompletion with 28510 gene symbols</strong>.  This short web form <a href="http://li1077-179.members.linode.com/gene.html">http://li1077-179.members.linode.com/gene.html</a> selects and displays candidate gene symbols that you start typing with auto-completion.  It selects candidate gene symbols from a long list of 28510 genes (gathered from UCSC's knownCanonical table in 2016) and uses Lea Verou's excellent Awesomplete library.   <a href="https://github.com/botanyhelp/Agemakeroffline/blob/master/gene.html">(code on github)</a></p>
</li>
<li>
<p><strong>UCSC Genome Browser dynamic custom track documentation</strong>.  Adding dynamic javascript to a UCSC genome browser custom track documentation file is possible.  <a href="https://github.com/botanyhelp/Gbjavascript">(code on github)</a></p>
</li>
<li>
<p><strong>Agemaker age calculating offline web app</strong>.  <a href="http://li1077-179.members.linode.com/agemaker.html">http://li1077-179.members.linode.com/agemaker.html</a> The application logic is written in Javascript and uses the entered birthdate to calculate the current age to one decimal place (for data entry).  This application is an offline web app which means that it will continue to work even after your Internet connection dies (on all modern browers).  The javascript, HTML5 and everything is in one file, agemaker.html:  <a href="https://github.com/botanyhelp/Agemakeroffline">(code on github)</a></p>
</li>
</ul>
<h3 id="perl">Perl</h3>
<ul>
<li><strong>Wisconsin Herbarium on Perl</strong>.  <!--TTT20210318 <a href="http://www.botany.wisc.edu/wisflora/">http://www.botany.wisc.edu/wisflora/</a>, -->  <a href="https://github.com/botanyhelp/Herbperl">(code on github)</a></li>
</ul>
<h3 id="php">Php</h3>
<ul>
<li>
<p><strong>Laravelcarriers</strong>.  Simple writable database program runnning on Laravel php web framework.  <a href="https://github.com/botanyhelp/Laravelcarriers">(code on github)</a></p>
</li>
<li>
<p><strong>Zipper</strong>.  zipper.php is a php program that creates zip archives.  Requires ZipArchive class be available: http://www.php.net/manual/en/class.ziparchive.php  <a href="https://github.com/botanyhelp/Zipper">(code on github)</a></p>
</li>
<li>
<p><strong>Phpuploader</strong>.  PHP script to upload files to web server.  <a href="https://github.com/botanyhelp/Phpuploader">(code on github)</a></p>
</li>
</ul>
<h3 id="python">Python</h3>
<ul>
<li>
<p><strong>Google App Engine</strong>: Each of the GAE apps are written in python</p>
<ul>
<!--TTT20210318
<li>
<p><a href="http://wisconsinherbarium.appspot.com/">http://wisconsinherbarium.appspot.com/</a></p>
</li>
-->
<li>
<p><a href="http://fixamfm.appspot.com/">http://fixamfm.appspot.com/</a></p>
</li>
<li>
<p><a href="http://fixwdc.appspot.com/">http://fixwdc.appspot.com/</a></p>
</li>
</ul>
</li>
<li>
<p><strong>Albumartmaker</strong>.  Creates new 300x300 PNG file in the current directory: test.png.  Requires modules Image, ImageDraw, and texwrap.  Will use ImageFont if you have them (/usr/share/fonts/bitstream-vera/Vera.ttf) <a href="https://github.com/botanyhelp/Albumartmaker">(code on github)</a></p>
</li>
<li>
<p><strong>Searchyumrepos</strong>.  Search yum software repositories for your list of software. <a href="https://github.com/botanyhelp/Searchyumrepos">(code on github)</a></p>
</li>
<li>
<p><strong>Gmailer</strong>.  Short python script to use gmail to send email, <a href="https://github.com/botanyhelp/Gmailer">(code on github)</a></p>
</li>
<li>
<p><strong>Ucscdiskspacer</strong>.  Program to report total size, in bytes, of UCSC genome browser databases, <a href="https://github.com/botanyhelp/Ucscdiskspacer">(code on github)</a>.  Run like this to learn the size of the total size of a zebrafish and human genome download <strong>python ucsc_disk_spacer.py danRer5 hg19</strong></p>
</li>
<li>
<p><strong>Ucscquery</strong>.  Python standalone program to query UCSC Genome Browser Mysql database, requires module MySQLdb, requires internet access, <a href="https://github.com/botanyhelp/Ucscquery">(code on github)</a>.  This program performs steps that can be run manually with a mysql client:</p>
</li>
</ul>
<pre><code>mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A
mysql&gt; use mm9;
Database changed
mysql&gt; select * from knownCanonical;
</code></pre>

<ul>
<li>
<p><strong>Botitwebmaker</strong>.  Python program that generates HTML web site from a directory of image files.  Generated the HTML found here: <a href="http://botit.botany.wisc.edu/">http://botit.botany.wisc.edu/</a>,  <a href="https://github.com/botanyhelp/Botitwebmaker">(code on github)</a></p>
</li>
<li>
<p><strong>QRCode</strong>.  <a href="https://github.com/botanyhelp/Qrcode">(code on github)</a>, see <a href="#ruby">Ruby</a> section for details.</p>
</li>
</ul>
<h3 id="r">R</h3>
<ul>
<li>
<p><strong>R programming certificates</strong>.  See <a href="#certificates">certificates earned</a> section. </p>
</li>
<li>
<p><strong>R package Rufosightings</strong>.  Provides UFO sightings data as an R dataframe and simple functions.  Use R to install and run it like this: </p>
</li>
</ul>
<pre><code class="r">library(devtools)
install_github(&quot;botanyhelp/Rufosightings&quot;)
library(Rufosightings)
ufo_byexactdate()
ufo_byexactdate(20100704)
dim(Rufosightings)
Rufosightings[Rufosightings$sighted_at_year==&quot;1950&quot;,]
Rufosightings[Rufosightings$sighted_at_year==&quot;1942&quot;,]$description
Rufosightings[Rufosightings$sighted_at_year==&quot;1942&quot;,]$duration
Rufosightings[Rufosightings$sighted_at_year==&quot;1942&quot;,]$shape
levels(Rufosightings$shape)
Rufosightings[Rufosightings$location_city==&quot; Slickville&quot;,]
library(dplyr)
glimpse(Rufosightings)
select(Rufosightings,c(location_city,shape,description)) %&gt;% filter(location_city==&quot; Miami&quot;) %&gt;% filter(shape==&quot; cigar&quot;)
</code></pre>

<ul>
<li><strong>R package Remdat</strong>.  Provides <a href="./www.emdat.be">www.emdat.be</a> disasters data as an R dataframe.  Use R to install and run it like this: </li>
</ul>
<pre><code class="r">library(devtools)
install_github(&quot;botanyhelp/Remdata&quot;)
library(Remdat)
head(Remdat)
</code></pre>

<ul>
<li>
<p><strong>R Shiny Web Application</strong>.  Uses the Remdat package for data to explore in the web application. </p>
<ul>
<li><a href="http://li1077-179.members.linode.com:8787/">Disasters Worldwide Shiny App http://li1077-179.members.linode.com:8787/)</a></li>
</ul>
</li>
</ul>
<h3 id="ruby">Ruby</h3>
<ul>
<li>
<p><strong>Herokucarriers</strong>.  Ruby on Rails web app writing to Postgres database all on Heroku  <a href="https://github.com/botanyhelp/Herokucarriers">(code on github)</a></p>
</li>
<li>
<p><strong>QRCode</strong>.  This ruby code creates png formatted images of the QR codes. It <strong>also</strong> dumps text which can be used to create an HTML/CSS version of the QR code.  After running the ruby program <strong>with the included example data</strong>, six new 300x300 PNG-formatted images will be created in the current directory. The output HTML from the python can be dropped right into any HTML file and will display the QR codes as HTML.  The workflow requires python and the ruby requires some modules that are probably not present.  Install them with gem like this and run the whole thing like this:</p>
</li>
</ul>
<pre><code>gem install rqrcode
gem install rqrcode_png
ruby qrHTMLandPNGmaker.rb &gt;&gt; rb.txt
python qrHTMLmaker.py &gt;&gt; qr_codes.html
</code></pre>

<h3 id="web-content">Web Content</h3>
<ul>
<li>
<p><strong>Multilingual and right to left HTML5</strong>.  Displays the hebrew language and instructs the browser to present it using a right-to-left reading orientation which is proper for right to left languages like hebrew.  Notice in your browser that it <strong>selects</strong> the text using the <strong>right to left</strong> orientation. <a href="http://li1077-179.members.linode.com/hebrew.html">http://li1077-179.members.linode.com/hebrew.html</a>,  <a href="https://github.com/botanyhelp/Agemakeroffline/blob/master/hebrew.html">(code on github)</a></p>
</li>
<li>
<p><strong>Two languages on one page</strong>.  One page of HTML, two languages: <a href="http://li1077-179.members.linode.com/russian.html">http://li1077-179.members.linode.com/russian.html</a>,  <a href="https://github.com/botanyhelp/Agemakeroffline/blob/master/russian.html">(code on github)</a></p>
</li>
<li>
<p><strong>Offline web application</strong>, Agemaker age calculating web page. This web application is an <strong>offline web app</strong> such that, when you revisit this URL later, <strong>when you have no Internet connection</strong>, it still works (on all modern web browsers). <a href="http://li1077-179.members.linode.com/agemaker.html">http://li1077-179.members.linode.com/agemaker.html</a>,  <a href="https://github.com/botanyhelp/Agemakeroffline">(code on github)</a></p>
</li>
</ul>

<h3 id="databases">Databases</h3>
<ul>
<li>
<p><strong>Google Data Store</strong>. These three App Engine web apps use the Google Data Store for persistence:</p>
<ul>
<li>
<p><a href="http://fixwdc.appspot.com/">http://fixwdc.appspot.com/</a></p>
</li>
<li>
<p><a href="http://fixamfm.appspot.com/">http://fixamfm.appspot.com/</a>  <a href="https://github.com/botanyhelp/fixamfm">(code on github)</a></p>
</li>
<li>
<p><!--TTT20210318<a href="http://wisconsinherbarium.appspot.com/">http://wisconsinherbarium.appspot.com/</a>-->Wisconsin Herbarium  <a href="https://github.com/botanyhelp/Herbpythonappengine">(code on github)</a></p>
</li>
</ul>
</li>
<li>
<p><strong>Postgres</strong>.  Rails applications running on the free tier at Heroku are required to use Postgres database,  <a href="https://github.com/botanyhelp/Herokucarriers/">(code on github)</a></p>
</li>
<li>
<p><strong>MySQL</strong>.<!--TTT20210318 <a href="http://www.botany.wisc.edu/wisflora/">Wisconsin Herbarium</a>-->Wisconsin Herbarium <a href="https://github.com/botanyhelp/Herbperl">(code on github)</a></li></p>
</li>
<li>
<p><strong>Sqlite</strong>.  A custom sqlite database is at the heart of over half of the android apps on this page: <a href="https://play.google.com/store/apps/developer?id=Thomas+Maher">https://play.google.com/store/apps/developer?id=Thomas+Maher</a></p>
</li>
</ul>

<h3 id="cloud">Cloud</h3>
<ul>
<li>
<p><strong>Google App Engine </strong>. Three python web applications on Google App Engine infrastructure:</p>
<ul>
 <li><!--TTT20210318<a href="http://wisconsinherbarium.appspot.com/">Wisconsin Herbarium</a>-->Wisconsin Herbarium  <a href="https://github.com/botanyhelp/Herbpythonappengine">(code on github)</a></li>
<li><a href="http://fixamfm.appspot.com/">Fix AM FM</a>  <a href="https://github.com/botanyhelp/fixamfm">(code on github)</a></li>
<li><a href="http://fixwdc.appspot.com/">Fix WDC</a></li>
</ul>
</li>
<li>
<p><strong>Heroku</strong>.  Ruby on Rails web application running on Heroku writing to Heroku Postgres database <a href="https://github.com/botanyhelp/Herokucarriers">(code on github)</a></p>
</li>
<li>
<p><strong>AWS</strong>.  <a href="https://www.credly.com/badges/aa47484c-cdf8-4aa3-b515-34dc6ec718a9">Certified Cloud Practitioner</a></p>
</li>
<li>
<p><strong>AWS Skill Builder Course Completions</strong>.  
</li>
<ul>
 <li>Introdution to AWS CodeBuild</li>
 <li>Introdution to AWS Scaling</li>
 <li>Introdution to EC2 Auto Scaling</li>
 <li>Introdution to Amazon Elastic Load Balancer - Application</li>
 <li>Introdution to Amazon Route 53</li>
 <li>Introduction to Containers</li>
 <li>Introdution to Amazon Elastic Container Registry</li>
 <li>Amazon Elastic Container Service (ECS) Primer</li>
 <li>Introdution to Amazon Elastic Container Service</li>
 <li>Managing application lifecycle in Amazon Elastic Container Service (ECS)</li>
 <li>Amazon Elastic Kubernetes Service (EKS) Primer</li>
 <li>AWS Application Services Overview</li>
 <li>Introdution to AWS IoT</li>
 <li>Configure and Deploy AWS PrivateLink</li>
 <li>Introdution to Amazon API Gateway</li>
 <li>Introdution to Dedicated Instance</li>
 <li>Introdution to Amazon EC2 Systems Manager</li>
 <li>Deploying and Scaling Applications with Amazon Lightsail</li>
 <li>Amazon RDS Service Introdution </li>
 <li>Amazon Aurora Service Primer</li>
 <li>Amazon Aurora Service Introdution </li>
 <li>Amazon Aurora MySQL - Basics</li>
</ul>
</ul>
</div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="./js/jquery-1.10.2.min.js"></script>
        <script src="./js/bootstrap-3.0.3.min.js"></script>
        <script src="./js/highlight.pack.js"></script>
        <script>var base_url = '.';</script>
        <script data-main="./mkdocs/js/search.js" src="./mkdocs/js/require.js"></script>
        <script src="./js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>

<!--
MkDocs version : 0.14.0
Build Date UTC : 2017-07-14 12:26:20.524852
-->
